department-of-veterans-affairs/caseflow
# Update connect_vbms for change to preprod certificate

Connects https://github.com/department-of-veterans-affairs/connect_vbms/pull/197

### Description
Bumps the connect_vbms gem to [08735e11](https://github.com/department-of-veterans-affairs/connect_vbms/commit/08735e11cd9fc196c8f889093580b9054fd2f094). 
This version allows an empty value for the `CONNECT_VBMS_KEYPASS` environment variable. This is an approach for dealing with the certificate update VBMS has been rolling out across its environments; while some use a key password, the preprod cert package does not use a key password (equivalent to an empty string). 

### Acceptance Criteria 
- [x] Query on VBMS FindDocumentSeriesReference returns non-error response

### Testing Plan
1. Deploy branch to preprod.
Jenkins settings:
- branch: `refs/heads/lisac/vbms-preprod-cert-update`
- env variables:
`APP_VERSION=lisac/vbms-preprod-cert-update`
`DEPLOY_BRANCH=lisac/vbms-preprod-cert-upgrade`

2. Get sample VBMS file numbers from credstash.
`credstash -t appeals-credstash get caseflow.monitor.preprod.target_file_num`
(from [monitor](https://github.com/department-of-veterans-affairs/appeals-deployment/blob/546d413c07f60e27878ff91c376afe41a20ed37d/ansible/roles/caseflow-monitor/defaults/main.yml))

3. Simulate a VBMS request in the rails console of a preprod app server.

```
$ cd /opt/<app directory>/src
$ sudo su
$ source ../<app>_env.sh   
$ bin/rails c
> require 'vbms'
> request = VBMS::Requests::FindDocumentSeriesReference.new("<filenumber>")
> client = VBMS::Client.from_env_vars(env_name: ENV["CONNECT_VBMS_ENV"], use_forward_proxy:false)
> res = client.send_request(request)
```
  where the `<filenumber>` can be retrieved from the shell script `/opt/<app dir>/<app>_env.sh`

4. Verify the response content returns typical content and does not report a certificate error (example of the latter: `HTTPI::SSLError (SSL_connect returned=1 errno=0 state=error: certificate verify failed (unable to get local issuer certificate))`)
